modules/@apostrophecms/admin-bar/ui/apos/components/TheAposAdminBar.vue

```diff
@@ -394,12 +487,29 @@ export default {
     });
 
     if (this.editMode) {
-      // The page always initially loads with fully rendered content,
-      // so refetch the content with the area placeholders and data instead
-      this.refresh();
+      // Watch out for legacy situations where edit mode is active
+      // but we are not in draft
+      if (this.draftMode !== 'draft') {
+        // Also refreshes
+        this.switchDraftMode('draft');
+      } else {
+        // The page always initially loads with fully rendered content,
+        // so refetch the content with the area placeholders and data instead
+        this.refresh();
+      }
     }
   },
   methods: {
+    revertPublishedToPreviousFromNotification() {
+      if (this.canRevertPublishedToPrevious) {
+        this.revertPublishedToPrevious();
+      } else {
+        apos.notify('Cannot revert because state has changed', {
+          type: 'error',
+          dismiss: false
+        });
```

Comment:
@boutell better error message or general approach?

Reply:
I guess the best possible approach would be to dismiss the notification as soon as the admin bar is aware there's change, but I think that could be a follow-on given how briefly this button appears to begin with.